array_sparse.go

```diff
@@ -499,6 +499,33 @@ func (a *sparseArrayObject) exportToArrayOrSlice(dst reflect.Value, typ reflect.
 	return a.baseObject.exportToArrayOrSlice(dst, typ, ctx)
 }
 
+// estimateMemUsage helps calculating mem usage for large sparse arrays.
+// It will sample the array and use those samples to estimate the total
+// mem usage.
+func (a *sparseArrayObject) estimateMemUsage(ctx *MemUsageContext) (estimate uint64, err error) {
+	var samplesVisited, memUsage uint64
+	totalItems := len(a.items)
+	if totalItems == 0 {
+		return memUsage, nil
+	}
+	sampleSize := totalItems / 10
+
+	// grabbing one sample every "sampleSize" to provide consistent
+	// memory usage across function executions
+	for i := 0; i < totalItems; i += sampleSize {
+		v := a.items[i].value
+		inc, err := v.MemUsage(ctx)
+		samplesVisited += 1
```

Comment:
[q] Can `samplesVisited` just be total items? The only time it's not is on error which I imagine we care less about.

Reply:
It cannot as given for example 100 total items, we sample every 10 so the samples visited would be 10 in this case.

Reply:
oh I see you're not incrementing by 1. Got it.